### PR TITLE
feat: save updates to common-storage service

### PR DESCRIPTION
**Pull request by CL-SHLOMIKONCHA**

### CODE REVIEW

File: src/exportImage.py [hunk 1]
```diff
@@ -2,28 +2,64 @@
...
 
 class ExportImage:
     def __init__(self):
         self.logger = Logger()
         self.__config = read_config()
 
-    def export(self, offset, bbox, filename, url):
+    def export(self, offset, bbox, filename, url, taskid):
         try:
+            es_obj = { "taskId": taskid, "filename": filename, "link": "/test/link"}
```
Comment (asafMasa) [inline src/exportImage.py]: change "link" to URI (full path to created file)

File: src/exportImage.py [hunk 2]
```diff
@@ -2,28 +2,64 @@
...
+                    "link": ''
+
+                }
+            }
+            if percent == 100:
+                doc["body"]["status"] = 'completed'
+                doc["body"]["link"] = unknown["link"]
+
+            requests.post(url=url, data=json.dumps(doc), headers=headers)
+            self.logger.info(f'Task Id "{unknown["taskId"]}" Updated database with progress: {percent}')
```
Comment (asafMasa) [inline src/exportImage.py]: debug not info

Comment (asafMasa) [reply src/exportImage.py]: put log before post

File: src/exportImage.py [hunk 3]
```diff
@@ -2,28 +2,64 @@
...
+            return percent
+        except ConnectionError as ce:
+            self.logger.error(f'Database connection failed: {ce}')
+            raise ce
+        except Exception as e:
+            doc["body"]["status"] = 'failed'
+            requests.post(url=url, data=json.dumps(doc), headers=headers)
+            self.logger.error(f'Task Id "{unknown["taskId"]}" Failed Update database: {e}')
```
Comment (asafMasa) [inline src/exportImage.py]: Failed **to** update

File: src/exportImage.py [hunk 4]
```diff
@@ -2,28 +2,64 @@
...
+            raise e
```
Comment (asafMasa) [inline src/exportImage.py]: we don't want to stop the process if progress update fails

File: src/exportImage.py [hunk 5]
```diff
@@ -2,28 +2,64 @@
...
+                doc["body"]["link"] = unknown["link"]
+
+            requests.post(url=url, data=json.dumps(doc), headers=headers)
+            self.logger.info(f'Task Id "{unknown["taskId"]}" Updated database with progress: {percent}')
+            return percent
+        except ConnectionError as ce:
+            self.logger.error(f'Database connection failed: {ce}')
+            raise ce
+        except Exception as e:
+            doc["body"]["status"] = 'failed'
```
Comment (asafMasa) [inline src/exportImage.py]: progress failed not all the process

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/exportImage.py (hunk 5 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in 3da7436

File: src/exportImage.py (hunk 4 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in 3da7436

File: src/exportImage.py (hunk 3 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in 3da7436

File: src/exportImage.py (hunk 2 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in 3da7436

File: src/exportImage.py (hunk 1 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in cb74980

File: src/exportImage.py [hunk 6]
```diff
@@ -25,41 +29,46 @@ def export(self, offset, bbox, filename, url, taskid):
             return result
         except Exception as e:
             self.logger.error(f'Error occurred while exporting: {e}.')
+            doc = {
+                "body": {
+                    "taskId": taskid,
+                    "status": "failed",
+                    "datetime": str(datetime.now()),
```
Comment (asafMasa) [inline src/exportImage.py]: update the "lastUpdateTime"

File: src/exportImage.py [hunk 7]
```diff
@@ -25,41 +29,46 @@ def export(self, offset, bbox, filename, url, taskid):
...
-        index = self.__config["es"]["index"]
-        hostip = self.__config["es"]["host_ip"]
-        port = self.__config["es"]["port"]
+        percent = floor(complete * 100)
+        doc = {
+            "body": {
+                "taskId": unknown["taskId"],
+                "status": "in-progress",
+                "progress": percent,
+                "datetime": str(datetime.now()),
```
Comment (asafMasa) [inline src/exportImage.py]: update the "lastUpdateTime"

File: src/exportImage.py [hunk 8]
```diff
@@ -25,41 +29,46 @@ def export(self, offset, bbox, filename, url, taskid):
...
 
     def progress_callback(self, complete, message, unknown):
-        index = self.__config["es"]["index"]
-        hostip = self.__config["es"]["host_ip"]
-        port = self.__config["es"]["port"]
+        percent = floor(complete * 100)
+        doc = {
+            "body": {
+                "taskId": unknown["taskId"],
+                "status": "in-progress",
```
Comment (asafMasa) [inline src/exportImage.py]: can you create an enum for the statuses?
ask @alebinson  what fields he uses in order to translate it correctly in the client

File: src/exportImage.py [hunk 9]
```diff
@@ -25,41 +29,46 @@ def export(self, offset, bbox, filename, url, taskid):
...
+                "progress": percent,
+                "datetime": str(datetime.now()),
+                "filename": unknown["filename"],
+                "link": ''
```
Comment (asafMasa) [inline src/exportImage.py]: why do you need to update it here?

File: src/exportImage.py [hunk 10]
```diff
@@ -25,41 +29,46 @@ def export(self, offset, bbox, filename, url, taskid):
...
+        }
 
-        url = f'http://{hostip}:{port}/indexes/{index}/document'
+        if percent == 100:
+            link = f'{self.__config["input_output"]["folder_path"]}/{unknown["filename"]}.gpkg'
+            doc["body"]["status"] = 'completed'
+            doc["body"]["link"] = link
 
+        self.update_db(doc)
+        return percent
```
Comment (asafMasa) [inline src/exportImage.py]: why do you need to return it?

**Review by asafMasa — CHANGES_REQUESTED** [review]

File: src/exportImage.py (hunk 7 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in d7a4411

File: src/exportImage.py (hunk 6 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in d7a4411

File: src/exportImage.py (hunk 8 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in 79c5e7e

File: src/exportImage.py (hunk 10 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in 5cf9860

File: src/exportImage.py (hunk 9 above)
Comment (CL-SHLOMIKONCHA) [reply src/exportImage.py]: fixed in 877e76c

**Review by asafMasa — APPROVED** [review]

**Pull request merged**